src/core/net/dns_headers.hpp

```diff
@@ -1402,6 +1736,33 @@ class SigRecord : public ResourceRecord, public Clearable<SigRecord>
      */
     const uint8_t *GetRecordData(void) const { return reinterpret_cast<const uint8_t *>(&mTypeCovered); }
 
+    /**
+     * This method parses and reads the SIG singer name from a message.
+     *
+     * @param[in]     aMessage          The message to read from. `aMessage.GetOffset()` MUST point to the start of DNS
+     *                                  header.
+     * @param[inout]  aOffset           On input, the offset in @p aMessage to start of signer name field.
+     *                                  On exit when successfully read, @p aOffset is updated to point to the byte
+     *                                  after the name field (i.e. start of signature field).
+     * @param[out]    aNameBuffer       A pointer to a char array to output the read name as a null-terminated C string
+     *                                  (MUST NOT be nullptr).
+     * @param[in]     aNameBufferSize   The size of @p aNameBuffer.
+     *
+     * @retval OT_ERROR_NONE            The name was read successfully. @p aOffset and @p aNameBuffer are updated.
+     * @retval OT_ERROR_PARSE           The SIG record in @p aMessage could not be parsed (invalid format).
+     * @retval OT_ERROR_NO_BUFS         Name could not fit in @p aNameBufferSize chars.
+     *
+     */
+    otError ReadSignerName(const Message &aMessage,
+                           uint16_t &     aOffset,
+                           char *         aNameBuffer,
+                           uint16_t       aNameBufferSize) const
+    {
+        return ResourceRecord::ReadName(aMessage, aOffset, /* aStartOffset */ aOffset - sizeof(SigRecord), aNameBuffer,
+                                        aNameBufferSize,
+                                        /* aSkipRecord */ false);
```

Comment:
It is well-documented above, but at first glance, this seems inconsistent with the other `Read*Name()` methods.

Reply:
The main difference is that in the case of SIG record `ReadSignerName()`, we still have the signature field after the name, so it updates the offset to end of the name to allow caller to read the signature field.

But in the case of PTR and SRV record the name is the last field in the record (last field we know and care about), so after reading the name we update offset to skip over the entire record.

Also I am treating the DNS records, similar to how we parse TLVs, in the sense that if there are any remaining fields/bytes in the record after the fields that we know and read, we just ignore them (this is to allow future extendibility of the record definitions). So in case of PTR/SRV, we do not assume that the offset after the name would be the same as the end of the record (we check and ensure that the name is in within the record data but when we want to skip over the record, we use the record data length directly).